Issue while compiling for Raspberry 4 B

While compiling for Raspberry 4 B there is a problem with this:

error[E0308]: mismatched types
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/wpactrl-0.4.0/src/wpactrl.rs:124:25
    |
124 |                 tv_sec: duration.as_secs() as i64,
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/wpactrl-0.4.0/src/wpactrl.rs:125:26
    |
125 |                 tv_usec: duration.subsec_micros() as i64,
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

Changing the i64 to i32 or isize in the source solves the issue, but the problem is the project which uses this is will be automatically installed/updated by a third service on multiple raspberries so it always pulls the code from the crate repository and would always cause this bug on every new install/update.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -5,7 +5,7 @@ edition = "2021"
 license = "MIT"
 license-file = "LICENSE"
 repository = "https://github.com/sauyon/wpa-ctrl-rs"
-version = "0.3.1"
+version = "0.4.0"
 authors = [ "Sauyon Lee <s@uyon.co>", "Steven Pease <peasteven@gmail.com>" ]
 
 [dependencies]

--- a/src/wpactrl.rs
+++ b/src/wpactrl.rs
@@ -121,8 +121,8 @@ fn select(fd: RawFd, duration: Duration) -> Result<bool> {
             std::ptr::null_mut(),
             std::ptr::null_mut(),
             &mut libc::timeval {
-                tv_sec: duration.as_secs() as i64,
-                tv_usec: duration.subsec_micros() as i64,
+                tv_sec: duration.as_secs().try_into().unwrap(),
+                tv_usec: duration.subsec_micros().try_into().unwrap(),
             },
         )
     };

